Tools/cases_generator/mypy.ini

```diff
@@ -0,0 +1,11 @@
+[mypy]
+files = Tools/cases_generator/
+pretty = True
+
+python_version = 3.10
+
+# and be strict!
+strict = True
+strict_concatenate = True
+enable_error_code = ignore-without-code,redundant-expr,truthy-bool
+warn_unreachable = True
```

Comment:
A _lot_ of the changes in this PR seem to be due to mypy flagging these intentionally unreachable `assert_never` branches as unreachable. Maybe we should disable this check for now and revert those changes in this PR, then consider in a separate PR whether it's worth it to enable this check?

```suggestion
```